platforms/documentation/docs/src/docs/userguide/platforms/jvm/toolchains.adoc

```diff
@@ -33,101 +33,214 @@ You can also control the JVM used to run Gradle itself using the <<gradle_daemon
+}
+----
+=====
 ====
 
-If the vendor you want to target is not a known vendor, you can still restrict the toolchain to those matching the `java.vendor` system property of the available toolchains.
+This does *not* guarantee the correct JDK is used and may cause issues when APIs have been backported to older Java versions.
 
-The following snippet uses filtering to include a subset of available toolchains.
-This example only includes toolchains whose `java.vendor` property contains the given match string.
-The matching is done in a case-insensitive manner.
+You should only use this method in cases where you need backward compatibility **but cannot use toolchains**.
 
-====
-include::sample[dir="snippets/java/toolchain-filters/kotlin/",files="build.gradle.kts[tags=toolchain-matching-vendor]"]
-include::sample[dir="snippets/java/toolchain-filters/groovy/",files="build.gradle[tags=toolchain-matching-vendor]"]
-====
+[[sec:java-home-toolchain]]
+=== 4. Environment variables (`JAVA_HOME`)
 
-[[sec:native_image]]
-=== Selecting toolchains that support GraalVM native image
+You can influence which JDK Gradle uses by setting the `JAVA_HOME` environment variable:
 
-If your project needs a toolchain with https://www.graalvm.org/latest/reference-manual/native-image/[GraalVM Native Image capability], you can configure the spec to request it:
+[source,bash]
+----
+export JAVA_HOME=/path/to/java17
+----
 
-====
-include::sample[dir="snippets/java/toolchain-filters/kotlin/",files="build.gradle.kts[tags=toolchain-native-image]"]
-include::sample[dir="snippets/java/toolchain-filters/groovy/",files="build.gradle[tags=toolchain-native-image]"]
-====
+This sets a default JDK for all Java-based tools on your system, including Gradle and Maven.
 
-Leaving that value unconfigured or set to `false` will not restrict the toolchain selection based on the Native Image capability.
-That means that a Native Image capable JDK can be selected if it matches the other criteria.
+WARNING: This does not override Gradle’s toolchain support or other project-specific configurations.
 
-=== Selecting toolchains by virtual machine implementation
+This approach is useful for legacy projects that do not use toolchains and expect a specific JDK to be active in the environment.
 
-If your project requires a specific implementation, you can filter based on the implementation as well.
-Currently available implementations to choose from are:
+However, since `JAVA_HOME` applies globally, it cannot be used to specify different JDK versions for different projects.
+It is more reliable to use <<sec:java-toolchains,toolchains>>, which allow setting the Java version at the project level.
 
-`VENDOR_SPECIFIC`::
-Acts as a placeholder and matches any implementation from any vendor (e.g. hotspot, zulu, ...)
-`J9`::
-Matches only virtual machine implementations using the OpenJ9/IBM J9 runtime engine.
+[[sec:ide-settings-toolchain]]
+=== 5. IDE settings
```

Comment:
❓  Should this section mention the Daemon Toolchains? IIRC, they should be picked up by the IDE automatically. Though, I am not sure if the support for this has already landed in a public IDE releases.

Reply:
I saw this in the IntelliJ IDEA EAP. Not sure if it made it in the main release yet. I will still add a note.